src/nvim/CMakeLists.txt

```diff
@@ -152,6 +154,21 @@ separate_arguments(C_FLAGS_ARRAY UNIX_COMMAND ${CMAKE_C_FLAGS})
 separate_arguments(C_FLAGS_${build_type}_ARRAY UNIX_COMMAND ${CMAKE_C_FLAGS_${build_type}})
 set(gen_cflags ${gen_cflags} ${C_FLAGS_${build_type}_ARRAY} ${C_FLAGS_ARRAY})
 
+function(get_preproc_output varname iname)
+  if(MSVC)
+    set(${varname} /P /Fi${iname} PARENT_SCOPE)
+  else()
+    set(${varname} -E -o ${iname} PARENT_SCOPE)
+  endif()
+endfunction()
+
+# NEOVIM_GENERATED_FOR_HEADERS: header files generated to be included in headers
+# NEOVIM_GENERATED_FOR_SOURCES: header files generated to be included in sources
+# NEOVIM_GENERATED_SOURCES: generated source files
+# These lists should be mutually exclusive
+# If a generated file is included in both of headers and sources,
+# the file for one should be duplicated for the other with different filepath.
```

Comment:
Removed the comment about a generated file which is included in both of sources and headers.